cterasdk/core/plans.py

```diff
@@ -143,3 +182,51 @@ def delete(self, name):
         except CTERAException as error:
             logging.getLogger().error("Plan deletion failed.")
             raise CTERAException('Plan deletion failed', error)
+
+
+class PlanAutoAssignPolicy(BaseCommand):
+
+    def get_policy(self):
+        """
+        Get plans auto assignment policy
+        """
+        return self._portal.execute('', 'getPlanAutoAssignmentRules')
+
+    def set_policy(self, rules, apply_default=None, default=None, apply_changes=True):
+        """
+        Set plans auto assignment policy
+
+        :param list[cterasdk.common.types.PolicyRule] rules: List of policy rules
+        :param bool,optional apply_default: If no match found, apply default plan. If not passed, the current config will be kept
+        :param str,optional default: Name of a plan to assign if no match found. Ignored unless the ``apply_default`` is set to ``True``
+        """
+        plans = [rule.value for rule in rules]
+        portal_plans = {plan.name: plan for plan in self._portal.plans.by_name(plans, ['baseObjectRef'])}
+
+        not_found = [plan for plan in plans if plan not in portal_plans.keys()]
+        if not_found:
+            logging.getLogger().error('Could not find one or more plans. %s', {'plans': not_found})
+            raise CTERAException('Could not find one or more plans', None, plans=not_found)
+
+        policy = self.get_policy()
+
+        if apply_default is False:
+            policy.defaultPlan = None
+        elif apply_default is True and default:
+            policy.defaultPlan = self._portal.plans.get(default, ['baseObjectRef']).baseObjectRef
+
+        policy_rules = []
+        for rule in rules:
+            policy_rule = PolicyRuleCoverter.convert(rule, 'PlanAutoAssignmentRule', 'plan')
```

Comment:
I don't understand the usage of the PolicyRuleConverter. The "convert" function will set the "plan" attribute to "rule.value" and then you override it. So, why not pass the correct value to begin with?

Reply:
Fixed.
`plans = set([rule.value for rule in rules])`
`if default:`
`    plans.add(default)`
`plans = list(plans)`
`portal_plans = {plan.name: plan for plan in self._portal.plans.by_name(plans, ['baseObjectRef'])}`

Then, when obtaining the `baseObjectRef`, we retrieve it from the dictionary.
`policy.defaultPlan = portal_plans.get(default).baseObjectRef`

Reply:
Fixed

Reply:
Since its a single line with append, you can change it to list comprehension:
policy_rules = [PolicyRuleCoverter.convert(...) for rule in rules]

Reply:
Done